lib/AST/Type.cpp

```diff
@@ -723,6 +723,32 @@ Type TypeBase::getUnlabeledType(ASTContext &Context) {
   return getStrippedType(Context, Type(this), /*stripLabels=*/true);
 }
 
+/// Remove argument labels from the function type.
+Type TypeBase::removeArgumentLabels(unsigned numArgumentLabels) {
```

Comment:
I know you're just refactoring this, but this seems like a less than ideal name for the parameter.

Reply:
I'm hoping this can go away altogether once we stop storing labels in the function type. For the most part the solver doesn't look at those labels at all, which is why it strips them. They might only matter in diagnostics, and we can come up with a bespoke mechanism to print an argument list with the labels provided out of band or something.